libs/ui/src/button.tsx

```diff
@@ -1,122 +1,353 @@
+
+export const ALL_BUTTON_VARIANTS = [
+  'danger',
+  'done',
+  'edit',
+  'next',
+  'previous',
+  'previousPrimary',
+  'primary',
+  'regular',
+  'secondary',
+  'settings',
+  'warning',
+] as const;
+
+export type ButtonVariant = typeof ALL_BUTTON_VARIANTS[number];
+
+type ClickHandler = () => void;
+type TypedClickHandler<T> = (value: T) => void;
 
-export interface ButtonInterface {
-  readonly danger?: boolean;
+export interface StyledButtonProps {
+  autoFocus?: boolean;
+  className?: string;
+  disabled?: boolean;
+  id?: string;
+  variant?: ButtonVariant;
+
+  /** @deprecated Place the button within a flex or grid container instead. */
   readonly fullWidth?: boolean;
+
+  /** @deprecated Incompatible with new VVSG size themes */
   readonly large?: boolean;
-  readonly noFocus?: boolean;
-  readonly noWrap?: boolean;
-  readonly primary?: boolean;
-  readonly primaryBlue?: boolean;
+
+  /** @deprecated Incompatible with new VVSG size themes */
   readonly small?: boolean;
-  readonly textAlign?: 'left' | 'center' | 'right';
-  readonly warning?: boolean;
 }
 
-export interface StyledButtonProps
-  extends ButtonInterface,
-    React.PropsWithoutRef<JSX.IntrinsicElements['button']> {}
+export type ButtonProps<T = undefined> = StyledButtonProps & {
+  children?: React.ReactNode;
+
+  /**
+   * @deprecated NOTE: Title tooltips are not accessible and should not be used
+   * for important information. Consider rendering the text on or near the
+   * button.
+   */
+  nonAccessibleTitle?: string;
+
+  /** @deprecated */
+  component?: StyledComponent<'button', DefaultTheme, StyledButtonProps, never>;
+} & ( // Require a matching `value` if the provided click handler expects a value.
+    | {
+        onPress: ClickHandler;
+        value?: never;
+      }
+    | {
+        onPress: TypedClickHandler<T>;
+        value: T;
+      }
+  );
+
+/** @deprecated Use {@link ButtonProps} instead. */
+export type ButtonInterface = Pick<
+  ButtonProps,
+  'fullWidth' | 'large' | 'small' | 'variant'
+>;
+
+interface VariantConfig {
+  color: keyof UiTheme['colors'];
+  icon?: React.ComponentType;
+  isSolidColor?: boolean;
+}
+
+const variantConfigs: Record<ButtonVariant, VariantConfig> = {
+  danger: { color: 'accentDanger', icon: Icons.Danger, isSolidColor: true },
+  done: { color: 'accentPrimary', icon: Icons.Done, isSolidColor: true },
+  edit: { color: 'foreground', icon: Icons.Edit },
+  next: { color: 'accentPrimary', icon: Icons.Next, isSolidColor: true },
+  previous: { color: 'foreground', icon: Icons.Previous },
+  previousPrimary: {
+    color: 'accentPrimary',
+    icon: Icons.Previous,
+    isSolidColor: true,
+  },
+  primary: { color: 'accentPrimary', isSolidColor: true },
+  regular: { color: 'foreground' },
+  secondary: { color: 'accentSecondary', isSolidColor: true },
+  settings: { color: 'foreground', icon: Icons.Settings },
+  warning: { color: 'accentWarning', icon: Icons.Warning, isSolidColor: true },
+};
```

Comment:
I appreciate this organization! I was wondering how I was going to keep track of all this.